File: backend/src/main/java/com/lms_erp/person/service/PersonDocumentService.java
package com.lms_erp.person.service;

import com.lms_erp.person.dto.DocumentFileResponse;
import com.lms_erp.person.dto.PersonDocumentDto;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface PersonDocumentService {

    // =====================================================
    // ADD DOCUMENT
    // =====================================================

    PersonDocumentDto addDocument(
            Long personId,
            PersonDocumentDto dto,
            MultipartFile file
    );

    // =====================================================
    // GET DOCUMENT METADATA
    // =====================================================

    List<PersonDocumentDto> getDocumentsByPersonId(
            Long personId
    );

    // =====================================================
    // UPDATE DOCUMENT
    // =====================================================

    PersonDocumentDto updateDocument(
            Long documentId,
            PersonDocumentDto dto,
            MultipartFile file
    );

    // =====================================================
    // DELETE DOCUMENT
    // =====================================================

    void deleteDocument(
            Long documentId
    );

    // =====================================================
    // GET ACTUAL FILE FROM DATABASE
    // =====================================================

    DocumentFileResponse getDocumentFile(
            Long documentId
    );

    boolean isOwner(Long personId);

    boolean isDocumentOwner(Long documentId);
}

File: lms-frontend/src/person/services/personDocumentService.js
import api from "../../api/axios";

const BASE_URL = "/api/person-documents";

const uploadDocument = (personId, documentData) => {

    return api.post(
        `${BASE_URL}/person/${personId}`,
        documentData
    );
};

const getDocuments = (personId) => {

    return api.get(
        `${BASE_URL}/person/${personId}`
    );
};

const updateDocument = (documentId, documentData) => {

    return api.put(
        `${BASE_URL}/${documentId}`,
        documentData
    );
};

const deleteDocument = (documentId) => {

    return api.delete(
        `${BASE_URL}/${documentId}`
    );
};

export default {
    uploadDocument,
    getDocuments,
    updateDocument,
    deleteDocument
};